rio_tiler/reader.py

```diff
@@ -261,8 +261,13 @@ def read(
 
         if unscale:
             data = data.astype("float32", casting="unsafe")
-            numpy.multiply(data, dataset.scales[0], out=data, casting="unsafe")
-            numpy.add(data, dataset.offsets[0], out=data, casting="unsafe")
+
+            # reshaped to match data
+            scales = numpy.array(dataset.scales).reshape((-1,) + (1,) * (len(data.shape[1:])))
+            offsets = numpy.array(dataset.offsets).reshape((-1,) + (1,) * (len(data.shape[1:])))
```

Comment:
```suggestion
            scales = numpy.array(dataset.scales).reshape(-1, 1, 1)
            offsets = numpy.array(dataset.offsets).reshape(-1, 1, 1)
```
should be enough, I don't see rasterio handling non 2D data